src/main/java/org/socialhistoryservices/delivery/reservation/controller/ReservationController.java

```diff
@@ -382,45 +435,119 @@ protected void autoPrint(final Reservation res) {
 
     /**
      * Mass delete reservations.
-     * @param req The HTTP request object.
+     *
+     * @param req     The HTTP request object.
      * @param checked The reservations marked for deletion.
      * @return The view to resolve.
      */
     @RequestMapping(value = "/batchprocess",
-                    method = RequestMethod.POST,
-                    params = "delete")
+            method = RequestMethod.POST,
+            params = "delete")
     @PreAuthorize("hasRole('ROLE_RESERVATION_DELETE')")
     public String batchProcessDelete(HttpServletRequest req,
-                                     @RequestParam(required=false) List<String>
+                                     @RequestParam(required = false) List<String>
                                              checked) {
-
         // Delete all the provided reservations
         if (checked != null) {
+            ArrayList<Integer> requestIds = new ArrayList<>();
+
+            for (BulkActionIds bulkActionIds : getIdsFromBulk(checked)) {
+                if (!requestIds.contains(bulkActionIds.getRequestId())) {
+                    requestIds.add(bulkActionIds.getRequestId());
+                }
+            }
+
+            // loop each reservation
+            for (Integer resId : requestIds) {
+                changeStatusOfHoldingsInReservation(resId, Holding.Status.RESERVED, Holding.Status.AVAILABLE);
```

Comment:
Code can be combined. But more importantly: don't do this on this level. See comment earlier.